### DISCUSSION TITLE
Maven, dev mode and resources

### DISCUSSION THREAD

**Discussion by lecogiteur** [quarkusio/quarkus#39737] [Community]
Hi,

I have a directory X in my test resources. When I launch `quarkus dev` I would want this directory is added to classpath.

I try this:
```xml
<profiles>
        <profile>
            <id>dev</id>
            <build>
                <resources>
                    <resource>
                        <directory>src/test/resources/X</directory>
                    </resource>
                </resources>
            </build>
        </profile>
    </profiles>
```

But it doesn't work.
I don't want this directory when I build the prod release. Only in dev mode.

How to do that?

**Comment by gsmet** [comment]
Not sure it's possible but @aloubyansky will know for sure.

**Reply by aloubyansky** [reply]
@lecogiteur could you try withsystem property `quarkus.bootstrap.effective-model-builder` set to true?

**Reply by lecogiteur** [reply]
thank you for your feedback.

I'm not sure to understand the purpose of `quarkus.bootstrap.effective-model-builder`. The doc is not clear for me.

I add it to my `application.properties` and set it to true. Nothing change. In fact, I wonder too how to launch maven plugin only in dev mode.
Another question the profile  `dev` quarkus is not linked to a profile maven ? (a profile maven "dev", for example? )

**Reply by aloubyansky** [reply]
Quarkus `dev` profile isn't linked to a Maven profile but you could add the `defaultGoal` to your profile like this
```
        <profile>
            <id>dev</id>
            <build>
                <defaultGoal>compile quarkus:dev</defaultGoal>
            </build>
        </profile>
```
and then launch dev mode with `mvn -Pdev`.

But back to your original question, I just looked and it doesn't look like it will work, unfortunately. You could create an issue though for this use-case.

**Reply by lecogiteur** [reply]
@aloubyansky , @gsmet I find this issue: https://github.com/quarkusio/quarkus/issues/23335

It's exactly what I want. My real case: I use liquibase extension and I have a changelog to populate the database only for dev mode. I can use `%dev.quarkus.liquibase.change-log` with a specific change logs which include changelogs of prod and some insert for dev mode.
The problem is like https://github.com/quarkusio/quarkus/issues/23335#issuecomment-1557703969 or https://github.com/quarkusio/quarkus/issues/23335#issuecomment-1026734383 . I must to put changelogs for dev mode in src/main/resource which would end up in the production jar.

There is some solutions?

**Reply by aloubyansky** [reply]
I think a Maven dev profile with custom resources should be the way to do that. But looking at the code, it doesn't look like it's going to work currently. I'll look into this.

**Reply by lecogiteur** [reply]
Thanks @aloubyansky

**Reply by lecogiteur** [reply]
Hi @aloubyansky ,

another way will be to exclude some directory from prod release. I try `quarkus.package.ignored-entries` or `quarkus.class-loading.removed-resources`. But it seems to not work. The changelog for dev mod is always include in prod release.

**Reply by aloubyansky** [reply]
Have you tried configuring removed-resources as a dev profile property?

**Reply by lecogiteur** [reply]
No on prod profile.

**Reply by lecogiteur** [reply]
I do this 

`quarkus.class-loading.removed-resources."my-group-project\:my-artefact"=db/changlog-dev`

Where `my-group-project` et `my-artefact` is the current project and the directory `db/changelog-dev` is in `src/main/resources`

**Reply by aloubyansky** [reply]
Right, my previous comment didn't make sense, need to wake up.
I guess what might work is if you create a Maven profile that's activeByDefault and set the remove-resources property in it. And then you keep your dev Maven profile to launch the app in dev mode, in this case the activeByDefault profile should be deactivated.

**Reply by lecogiteur** [reply]
Yes. The profile Maven work well. But it's could be a excellent enhancement to simply add or remove resources with quarkus profile.

**Comment by aloubyansky** [comment]
Actually, the following works
```
        <profile>
            <id>dev</id>
            <activation>
                <property>
                    <name>dev</name>
                </property>
            </activation>
            <build>
                <defaultGoal>compile quarkus:dev</defaultGoal>
                <resources>
                    <resource>
                        <directory>src/dev/resources</directory>
                    </resource>
                    <resource>
                        <directory>src/main/resources</directory>
                    </resource>
                </resources>
            </build>
        </profile>
```
Adding `src/dev/resources/META-INF/resources/dev.html` and launching the app with `mvn -Pdev`, I can access `localhost:8080/dev.html` and live coding in it works.

**Reply by aloubyansky** [reply]
The reason I thought it wouldn't work was that we initialize module source and resources paths from a "raw" POM model w/o interpolating active profiles but when we bootstrap for dev mode, the raw model isn't exactly raw, we use interpolated `<build>` part of it, so it works.
